pkg/plugin/builtin/registry.go

```diff
@@ -24,27 +25,29 @@ import (
 	postgres "github.com/conduitio/conduit-connector-postgres"
 	"github.com/conduitio/conduit-connector-protocol/cpluginv1"
 	s3 "github.com/conduitio/conduit-connector-s3"
-	"github.com/conduitio/conduit-connector-sdk"
+	sdk "github.com/conduitio/conduit-connector-sdk"
 	"github.com/conduitio/conduit/pkg/foundation/cerrors"
 	"github.com/conduitio/conduit/pkg/foundation/log"
 	"github.com/conduitio/conduit/pkg/plugin"
 	builtinv1 "github.com/conduitio/conduit/pkg/plugin/builtin/v1"
 )
 
 var (
-	DefaultDispenserFactories = []DispenserFactory{
-		sdkDispenserFactory(file.Connector),
-		sdkDispenserFactory(kafka.Connector),
-		sdkDispenserFactory(generator.Connector),
-		sdkDispenserFactory(s3.Connector),
-		sdkDispenserFactory(postgres.Connector),
-		sdkDispenserFactory(connLog.Connector),
+	// DefaultDispenserFactories contains default dispenser factories for
+	// built-in plugins. The key of the map is the import path of the module
+	// containing the connector implementation.
+	DefaultDispenserFactories = map[string]DispenserFactory{
+		"github.com/conduitio/conduit-connector-file":      sdkDispenserFactory(file.Connector),
+		"github.com/conduitio/conduit-connector-kafka":     sdkDispenserFactory(kafka.Connector),
+		"github.com/conduitio/conduit-connector-generator": sdkDispenserFactory(generator.Connector),
+		"github.com/conduitio/conduit-connector-s3":        sdkDispenserFactory(s3.Connector),
+		"github.com/conduitio/conduit-connector-postgres":  sdkDispenserFactory(postgres.Connector),
+		"github.com/conduitio/conduit-connector-log":       sdkDispenserFactory(connLog.Connector),
 	}
```

Comment:
Since all of our built-in connectors follow the same naming convention, we can determine the module name from the connector name.

Reply:
I was thinking of ways to determine the module (reflection, naming convention) and went with the dumb solution at the end. I'm not convinced it's worth changing it. I think your proposal would be a good fit if we were adding built-in plugins on a more regular basis, but for now I'd leave it like this.